paasta_tools/kubernetes_tools.py

```diff
@@ -1937,9 +1934,28 @@ def get_pod_anti_affinity(self) -> Optional[V1PodAntiAffinity]:
                         label_selector=label_selector,
                     )
                 )
+        return affinity_terms
+
+    def get_pod_anti_affinity(self) -> Optional[V1PodAntiAffinity]:
+        """
+        Converts the given anti-affinity on service and instance to pod
+        affinities with the "paasta.yelp.com" prefixed label selector
+        :return:
+        """
+
+        required_terms = self.get_pod_anti_affinity_terms(
+            config_key="anti_affinity",
+        )
+        preferred_terms = self.get_pod_anti_affinity_terms(
+            config_key="anti_affinity_preferred",
+        )
+
+        if required_terms is None and preferred_terms is None:
+            return None
 
         return V1PodAntiAffinity(
-            required_during_scheduling_ignored_during_execution=affinity_terms
+            required_during_scheduling_ignored_during_execution=required_terms,
+            preferred_during_scheduling_ignored_during_execution=preferred_terms,
```

Comment:
d'oh, preferred_during_scheduling_ignored_during_execution needs to be a list of V1WeightedPodAffinityTerm, not V1PodAffinityTerm. I'm gonna have to un-DRY this.

Reply:
heh, i guess that my assumption that there'd be a default weight and we could use the same class for both anti-affinities was wrong :p